### ISSUE TITLE
Certification program for remote storages that get highki

### ISSUE THREAD

**Issue by ramonguiu** [jaegertracing/jaeger#3826]
### Requirement

As a user I want to use a durable and scalable remote storage for Jaeger and I want to easily find out what options are available so that I can select the one that is a best fit for my needs.

### Problem

Currently, Jaeger provides a lot of visibility to four storages: in-memory, badger, ElasticSearch and Cassandra. The last two are remote storages that are deeply integrated into the code.

There are a number of other remote storage systems the community has integrated over the years that are currently very hard to discover and unclear what their benefits are and whether they have some level of support or not. An example of such a backend is Promscale which integrates via the remote gRPC API.

The majority of users are not using neither ElasticSearch not Cassandra even though I think using a durable remote storage would be recommended for most production use cases. I would think that a reason for that is that those backends may not be a good fit for many users. We think Promscale and other high quality remote storage systems would help the Jaeger community by providing them with other remote storage options that would fit their needs.

### Proposal

Create a certification program for all remote backends that lists a series of requirements that a remote backend needs to meet to be certified and promote the certified backends visibly in the documentation so they are easy to find out for users.

**Comment by yurishkuro** [comment]
> The majority of users are not using neither ElasticSearch not Cassandra

what do you base this on?

**Comment by ramonguiu** [comment]
> what do you base this on?
Mostly conversations we've had with Jaeger users and some folks in the tracing community.

Do you disagree? Do you think the majority of users are using Elasticsearch and Cassandra?

**Comment by yurishkuro** [comment]
I don't have any reliable data to claim one way or another. But the statement still doesn't ring true to me - if people are running a "serious" installation of Jaeger, then memory/Badger are not an option, and given the choice between supported backends (ES & Cass) and some bespoke solution, I would think the majority of people would use supported backends. We get many questions about ES, running with AWS ES. Anyway, this is just a side note.

**Comment by ramonguiu** [comment]
Thanks for sharing your insights. I've rephrased the description.

Let me know what we can do on our side to move this forward.

**Comment by yurishkuro** [comment]
For Promscale, I recommend:
  1. implement write path of remote API
  2. create a new test similar to https://github.com/jaegertracing/jaeger/blob/3d67ac86f53866332568b0f542265b00d86407fc/plugin/storage/integration/grpc_test.go#L189 that can be configured from cmd line and run with arbitrary remote backend

**Comment by ramonguiu** [comment]
Quick update on this:

1. Implementation of the write path of remote API is in progress: https://github.com/timescale/promscale/issues/1535
2. We have planned the work on allowing to run the tests from command line. I expect us to do the work this month.

**Comment by cevian** [comment]
@yurishkuro hey for item (2) do you recommend a new binary under `cmd/` or somewhere else? Does the name `remote-storage-tester` sound good? Or maybe `remote-storage-certification`? 

Thanks!

**Comment by yurishkuro** [comment]
@cevian I don't think we need a new binary, just a test suite similar to https://github.com/jaegertracing/jaeger/blob/3d67ac86f53866332568b0f542265b00d86407fc/plugin/storage/integration/grpc_test.go#L189

It can be run with `go test -v ...` so that the actual testing code has access to `testing.T` for assertions, etc.

**Comment by VineethReddy02** [comment]
@yurishkuro Promscale now supports the gRPC-based Jaeger write interface. With this now the Jaeger users can use Promscale for both ingesting and querying. 

Configuring Promscale as the backend in Jaeger collector/query is as simple as passing one CLI option i.e. `--grpc-storage.server=promscale:9202`.

**Comment by VineethReddy02** [comment]
@yurishkuro [here](https://github.com/VineethReddy02/jaeger-documentation.git) is the PR reflecting the same in the Jaeger docs.

**Comment by arajkumar** [comment]
> For Promscale, I recommend:
> 
> 1. implement write path of remote API
> 2. create a new test similar to https://github.com/jaegertracing/jaeger/blob/3d67ac86f53866332568b0f542265b00d86407fc/plugin/storage/integration/grpc_test.go#L189
>     that can be configured from cmd line and run with arbitrary remote backend

Hello @yurishkuro, Do you think #2(integration test for promscale) can be a part of the upstream jaeger repo?

**Comment by yurishkuro** [comment]
Seems odd to have it there if we're not going to execute it.

**Comment by ramonguiu** [comment]
@yurishkuro the work to enable any remote storage to self-certify using the Jaeger remote storage tests was merged in [this PR](https://github.com/jaegertracing/jaeger/pull/3944). 

We ran the tests against Promscale which revealed some failing tests that have now been fixed and would be included in the next release (going out in the next few days). It's been valuable to us.

As for next steps:
1. How do you suggest we document how the certification works so other remote storages can use it? Would you be interested in us writing a post in the Jaeger blog to announce the program?
2. Where do we document certified storage systems so they are easier to discover for the community?

**Comment by yurishkuro** [comment]
I don't have strong feelings how to do that, do you have suggestions?

I think it makes sense to add some README alongside of #3944 to describe how a non-core storage can be verified/tested. Then where ever we have "the list" we can have a link to CI job that illustrates passing of all the tests.

**Comment by ramonguiu** [comment]
That makes sense. We'll add a README to explain how this works. cc: @arajkumar 

We'll come with a suggestion about where we could place the list of certified storages with the link to the CI jobs showing all tests are passing.

**Comment by arajkumar** [comment]
@yurishkuro Created a PR to mention about storage certification.
https://github.com/jaegertracing/jaeger/pull/3985
